src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -159,6 +159,13 @@ internal ODataQueryContext QueryContext
         /// </summary>
         public ODataPath Path { get; set; }
 
+        internal Type Type { get; set; }
+
+        internal bool IsUntyped { get { return typeof(IEdmObject).IsAssignableFrom(Type); } }
```

Comment:
>return typeof(IEdmObject).IsAssignableFrom(Type) [](http://example.com/codeflow?start=40&length=48)

this is expensive. calculate this the first time it's requested and then cache until Type is reset.